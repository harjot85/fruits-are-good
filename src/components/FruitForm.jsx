import { Button, Form, FormGroup, Label, Input, Row, Col } from "reactstrap";
import React from "react";

const FruitForm = (props) => {
    const { onInputChange, onClickHandler, onKeyDown } = props;

    return (
        <>
            <Form>
                <FormGroup>
                    <Row>
                        <Col style={{textAlign: 'right'}}>
                            <Label style={{verticalAlign: 'sub'}} > Enter a fruit name</Label>
                        </Col>
                    
                        <Col>
                            <Input
                                type="text"
                                name="fruitName"
                                placeholder="Please enter a fruit name "
                                onChange={onInputChange}
                                onKeyDown={onKeyDown}
                                maxLength="25"
                            />
                        </Col>
                    
                        <Col>
                            <Button color="warning" id="submit" onClick={onClickHandler}>
                                Get Info
                            </Button>
                        </Col>
                    </Row>
                </FormGroup>
            </Form>
        </>
    );
};

export default FruitForm;
