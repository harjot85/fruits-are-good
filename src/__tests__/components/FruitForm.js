import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

// UI Components
import FruitForm from "../../components/FruitForm";
import { Button } from "reactstrap";

Enzyme.configure({ adapter: new Adapter() });

const mockProps = {
    onInputChange: jest.fn(),
    onClickHandler: jest.fn(),
    onKeyDown: jest.fn(),
};

let wrapper;

beforeEach(() => {
    wrapper = shallow(<FruitForm {...mockProps} />);
});

describe("<FruitForm />", () => {
    it("should match snapshot with props", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should render one <Button> component", () => {
        expect(wrapper.find(Button)).toHaveLength(1);
    });
});
