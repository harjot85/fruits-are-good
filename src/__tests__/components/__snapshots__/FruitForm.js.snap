// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FruitForm /> should match snapshot with props 1`] = `ShallowWrapper {}`;
